lib/std/fs.zig

```diff
@@ -451,6 +452,55 @@ pub const Dir = struct {
                 }
             }
 
+            fn nextDragonFly(self: *Self) !?Entry {
```

Comment:
What was different about DragonFlyBSD that nextBsd() didn't work?

Reply:
DragonFlyBSD doesn't use `dirent.d_reclen`, but instead uses this macro:

> #define	_DIRENT_RECLEN(namelen) \
>	((__offsetof(struct dirent, d_name) + (namelen) + 1 + 7) & ~7)

Reply:
```zig
pub const dirent = extern struct {
    d_fileno: c_ulong,
    d_namlen: u16,
    d_type: u8,
    d_unused1: u8,
    d_unused2: u32,
    d_name: [256]u8,

    fn reclen(self: dirent) usize {
        return (@byteOffsetOf(dirent, "d_name") + self.d_namlen + 1 + 7) & ~7;
    }
}; 
```

Then for the other BSDs where they have a `d_reclen`, add this:

```zig
    fn reclen(self: dirent) usize {
        return self.d_reclen;
    }
```

That's these files:

```
[nix-shell:~/Downloads/zig/build]$ grep -IR d_reclen ../lib/std/os/bits
../lib/std/os/bits/netbsd.zig:    d_reclen: u16,
../lib/std/os/bits/linux.zig:    d_reclen: u16,
../lib/std/os/bits/darwin.zig:    d_reclen: u16,
../lib/std/os/bits/freebsd.zig:    d_reclen: u16,
```

Reply:
Good call. I have pushed these changes

If any other changes are needed, please go ahead and make them